rp2040-hal/src/clocks/mod.rs

```diff
@@ -327,30 +483,29 @@ pll_usb_auxsource!(ReferenceClock, clk_ref_ctrl);
 gpin0_auxsource!(ReferenceClock, clk_ref_ctrl);
 gpin1_auxsource!(ReferenceClock, clk_ref_ctrl);
 
-int_division!(ReferenceClock, clk_ref_div, u8);
-
+division!(ReferenceClock, clk_ref_div);
 /// System Clock
 pub struct SystemClock {
     shared_dev: ShareableClocks,
 }
 impl SystemClock {
     /// WIP - Helper function to reset source (blocking)
-    pub fn reset_source_await(&self) {
+    pub fn reset_source_await(&mut self) {
         let shared_dev = unsafe { self.shared_dev.get() };
 
         shared_dev.clk_sys_ctrl.write(|w| {
             w.src().clear_bit();
             w
         });
 
-        self.await_select(0x1);
+        self.await_select(0x0);
     }
 
     /// WIP - Helper function to select new source (blocking)
     pub fn await_select(&self, clock: u8) {
         let shared_dev = unsafe { self.shared_dev.get() };
 
-        while shared_dev.clk_sys_selected.read().bits() != clock as u32 {
+        while (shared_dev.clk_sys_selected.read().bits() & 1 << clock) == 0 {
```

Comment:
Some parens around 1 << clock would be nice just so people don't have to remember whether << or & is evaluated first. Or assign it to a variable first, that would be even easier to read.

Reply:
Sorry, should have been more clear that I wanted
```
shared_dev.clk_sys_selected.read().bits() & (1 << clock)
```
here, to make it obvious that 1 << clock resolves before being bitwise AND'd